graphics/tinygl/light.cpp

```diff
@@ -237,7 +237,7 @@ void gl_shade_vertex(GLContext *c, GLVertex *v) {
 			d.Y = l->position.Y - v->ec.Y;
 			d.Z = l->position.Z - v->ec.Z;
 			dist = sqrt(d.X * d.X + d.Y * d.Y + d.Z * d.Z);
-			if (dist > 1E-3) {
```

Comment:
Sorry, missed that the dot product guarantees it indirectly. I withdraw my objection (and probably no need to fix the intermediate state either).